src/commons/util/mod.rs

```diff
@@ -1,16 +1,230 @@
 //! General utility modules for use all over the code base
+use std::cmp::Ordering;
+use std::fmt;
+use std::net::IpAddr;
+use std::str::FromStr;
+
+use serde::{de, Deserialize, Deserializer, Serialize, Serializer};
+
 use bytes::Bytes;
 use rpki::repository::crypto::DigestAlgorithm;
 use rpki::uri::{Https, Rsync};
-use std::net::IpAddr;
-use std::str::FromStr;
+
+use crate::constants::KRILL_VERSION;
 
 pub mod ext_serde;
 pub mod file;
 pub mod httpclient;
 pub mod softsigner;
 pub mod xml;
 
+
+//------------ KrillVersion --------------------------------------------------
+
+/// Defines a Krill version. Will 
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct KrillVersion {
+    major: u64,
+    minor: u64,
+    patch: u64,
+    release: KrillVersionRelease
```

Comment:
s/KrillVersionRelease/ReleaseType/ ? (version threw me off as we already express the version here in terms of major, minor, patch).

Reply:
Renamed to KrillVersionReleaseType. A bit wordy but okay and descriptive methinks.